Sources/PaymentsCore/Networking/APIClient.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+public class APIClient {
+    public typealias CorrelationId = String
+
+    public var urlSession: URLSession
+    public var environment: Environment
+
+    public init(urlSession: URLSession = .shared, environment: Environment) {
+        self.urlSession = urlSession
+        self.environment = environment
+    }
+
+    public func fetch<T: APIRequest>(
+        endpoint: T,
+        completion: @escaping (Result<T.ResponseType?, CoreError>, CorrelationId?) -> Void
+    ) {
+        guard let request = endpoint.toURLRequest(environment: environment) else {
+            completion(.failure(.noUrlRequest), nil)
+            return
+        }
+
+        let task = urlSession.dataTask(with: request) { data, response, error in
+            guard let response = response as? HTTPURLResponse else { return }
+            let correlationId = response.allHeaderFields["Paypal-Debug-Id"] as? String
+
+            let finish: (Result<T.ResponseType?, CoreError>) -> Void = { result in
+                /// For discussion:
+                /// When a network request to a PayPal API fails, we have some associated error information in the body data of the response.
+                /// This error data doesn't have the same format, but we should discuss and plan for how we want to surface this to the merchant
```

Comment:
I wonder if this is worth bringing up in a meeting to discuss with the API team. It would be great if we had an expected model to follow for all error cases so maybe this is an update that could be made on their end to formalize and error format?

Reply:
I'd love that (I think our GQL API works like that, and in NXO SDK we just have one error type to decode every errors => makes our lives much easier)
But at the same time, I think it'd be a big effort for the API team + considering Orders API is already live and there are merchants using it, I highly doubt it's gonna be viable... :(

Reply:
Yeh, the errors I think unfortunately will be far from uniform. They can also change without warning so our SDK should be ready for that.